contxt/cli/commands/iot.py

```diff
@@ -113,3 +115,193 @@ def data(
+        ]
+        reader = csv.DictReader(f, fieldnames=headers)
+
+        # skip the header
+        next(reader)
+
+        # create a bunch of field objects
+        for row in reader:
+            if row["Label"].replace(" ", "").startswith("-"):
+                label = row["equipment group"] + row["Label"]
+            else:
+                label = row["Label"]
+
+            if row["Field Descriptor"] in fields_descriptors_for_feed:
+                print(f'Already provisioned field with descriptor {row["Field Descriptor"]}')
+            else:
+                field_objs.append(
+                    Field(
+                        label=label,
+                        output_id=feed.id,
+                        field_descriptor=row["Field Descriptor"],
+                        units=row["Units"],
+                        field_human_name=row["Field Descriptor"],
+                        feed_key=feed_key,
+                        value_type=FieldValueType("numeric"),
+                        is_hidden=False,
+                    )
+                )
+
+            desired_groupings_by_field[row["Field Descriptor"]] = row["equipment group"]
+
+    # go provision the fields
+    for field_obj in field_objs:
+        clients.iot.provision_field_for_feed(feed.id, field_obj)
+        print(f"Provisioned: {field_obj.field_descriptor}")
+    print("Done provisioning the fields")
+
+    fields_for_feed = clients.iot.get_fields_for_feed(feed_id=feed.id)
+
+    # organize the fields by their grouping names
+    grouping_fields: Dict = {}
+    for field in fields_for_feed:
+        if field.field_descriptor in desired_groupings_by_field:
+            grouping_name = desired_groupings_by_field[field.field_descriptor]
+            if grouping_name.replace(" ", "") == "":
+                continue
+            if grouping_name not in grouping_fields:
+                grouping_fields[grouping_name] = []
+            grouping_fields[grouping_name].append(field.id)
+
+        # go through organized listing, create grouping if necessary, update fields for grouping
+        for grouping_name, field_ids in grouping_fields.items():
+            grouping = groupings_by_label.get(grouping_name)
+
+            if not grouping:
+                print(f"Creating new grouping: {grouping_name}")
+                grouping = clients.iot.create_grouping(
+                    facility_id=feed.facility_id,
+                    label=grouping_name,
+                    description=grouping_name,
+                    is_public=True,
+                    field_category_id=[],
+                )
+                print(grouping)
+
+            print(f"Setting {len(field_ids)} fields to grouping {grouping_name}")
+            clients.iot.set_fields_for_grouping(grouping.id, field_ids)
+
+        print("Successful!")
+
+
+@iot.command()
+@click.argument("feed_key", type=str)
+@click.pass_obj
+def create_worksheet(clients: Clients, feed_key) -> None:
```

Comment:
this whole command is unnecessary if you could simply retrieve unprovisioned fields